File: website/src/logic.js
import {create, remove, retrieve} from './backend';

export class Todo {
    constructor(uuid, name) {
        this.uuid = uuid;
        this.name = name;
    }
}

export function retrieveTodos() {
    return retrieve().catch(reason => {
        console.error('Retrieve Error:', reason);
        return Promise.reject('Encountered an error while retrieving todos.');
    });
}

export function addTodo(name) {
    return create(name).catch(reason => {
        console.error('Add Error:', reason);
        return Promise.reject('Encountered an error while adding a todo.');
    });
}

export function deleteTodo(uuid) {
    return remove(uuid).catch(reason => {
        console.error('Delete Error:', reason);
        return Promise.reject('Encountered an error while deleting a todo.');
    });
}

File: backend/backend/logic.py
from typing import List

from backend.storage import Storage
from backend.todos import Todo


class Logic:
    def __init__(self, storage: Storage):
        self.storage = storage

    def retrieve_todos(self) -> List[Todo]:
        return self.storage.retrieve_todos()

    def create_todo(self, todo: Todo):
        self.storage.persist_todo(todo)

    def delete_todo(self, uuid: str):
        self.storage.delete_todo(uuid)
